src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,126 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.ListCommand;
+import duke.command.LoadCommand;
+import duke.command.MarkCommand;
+import duke.command.MatchCommand;
+import duke.command.SaveCommand;
+import duke.command.UnmarkCommand;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * Deals with making sense of the user command.
+ */
+public class Parser {
+    /**
+     * Parses the user command and returns the corresponding command object.
+     *
+     * @param command The user command
+     * @return The corresponding command object
+     * @throws DukeException if the command is invalid
+     */
+    public static Command parse(String command) throws DukeException {
+        switch (command) {
+        case "bye":
+            return new ExitCommand();
+        case "list":
+            return new ListCommand();
+        case "mark":
+            return new MarkCommand();
+        case "unmark":
+            return new UnmarkCommand();
+        case "deadline":
+        case "todo":
+        case "event":
+            return new AddCommand();
+        case "delete":
+            return new DeleteCommand();
+        case "save":
+            return new SaveCommand();
+        case "load":
+            return new LoadCommand();
+        case "date":
+            return new MatchCommand();
+        default:
+            throw new DukeException("I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    /**
+     * Finds the String between first command and second command (if exist).
+     *
+     * @param input User input
+     * @param command First command
+     * @return String between first command and second command (if exist)
+     * @throws DukeException if the command is invalid
+     */
+    public static String findFirstCommand(String input, String command) throws DukeException {
+        int endOfCommand = input.indexOf("/");
+        int beginIndex = input.indexOf(command) + command.length() + 1;
+        if (beginIndex > input.length()) {
```

Comment:
```suggestion
boolean isIncorrectInput = beginIndex > input.length();
```
Similarly for line 81, where the expressions can be simplified